pkg/scheduler/queue/tenant_querier_assignment.go

```diff
@@ -286,91 +163,27 @@ func (tqa *tenantQuerierAssignments) removeTenant(tenantID TenantID) {
 	delete(tqa.tenantsByID, tenantID)
 }
 
-func (tqa *tenantQuerierAssignments) removeQuerierWorkerConn(conn *QuerierWorkerConn, now time.Time) (resharded bool) {
-	querier := tqa.queriersByID[conn.QuerierID]
-	if querier == nil || !querier.IsActive() {
-		panic("unexpected number of connections for querier")
-	}
-
-	if !conn.IsRegistered() {
-		panic("received request to deregister a querier-worker which was not already registered")
-	}
-	querier.RemoveWorkerConn(conn)
-	if querier.IsActive() {
-		// Querier still has active connections; it will not be removed, so no reshard occurs.
-		return false
-	}
-
-	// No more active connections. We can remove the querier only if
-	// the querier has sent a shutdown signal or if no forget delay is enabled.
-	if querier.shuttingDown || tqa.querierForgetDelay == 0 {
-		return tqa.removeQuerier(conn.QuerierID)
-	}
-
-	// No graceful shutdown has been notified yet, so we should track the current time
-	// so that we'll remove the querier as soon as we receive the graceful shutdown
-	// notification (if any) or once the threshold expires.
-	querier.disconnectedAt = now
-	return false
+// TODO (casie): Docstring
+// only adds querier to querierIDsSorted. Anywhere else a querier needs to be added should be handled by the component which manages that.
+func (tqa *tenantQuerierAssignments) addQuerier(querierID QuerierID) {
+	tqa.querierIDsSorted = append(tqa.querierIDsSorted, querierID)
+	sort.Sort(tqa.querierIDsSorted)
 }
 
-// removeQuerier deletes a querier from the tenant-querier assignments.
-// Returns true if tenant-querier reshard was triggered.
-func (tqa *tenantQuerierAssignments) removeQuerier(querierID QuerierID) (resharded bool) {
-	delete(tqa.queriersByID, querierID)
+// removeQueriers deletes an arbitrary number of queriers from the querier connection manager, and returns true if
+// recomputing tenant-querier sharding was necessary.
```

Comment:
imo "was necessary" leaves a little ambiguity about if it happened - let's just say if the tenant-querier sharding _was recomputed_.